### PR TITLE
feat: Scan code for security issues

### PR DESCRIPTION
**Pull request by ptiurin**
Adding security scan GH action. It's supposed to be run manually on a PR, at the same time as the integration tests. It also runs automatically on merge to master.
Using pipenv to build the project in order for Fossa to pick deep dependencies according to the [docs](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/python/python.md).

### CODE REVIEW

File: .github/workflows/security-scan.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout Code"
+        uses: actions/checkout@v2
+
+      - name: "Security Scan"
+        uses: firebolt-db/action-security-scan@main
+        with:
+          github-key: ${{ secrets.GITHUB_TOKEN }}
+          fossa-key: ${{ secrets.fossaApiKey }}
```
Comment (stepansergeevitch) [inline .github/workflows/security-scan.yml:20]: Is it possible to have this secret in upper snake case?
Like `secrets.FOSSA_API_KEY`

**Review by stepansergeevitch — APPROVED** [review]
Overall LGTM.
One issue is that currently sonar cloud seems to fail because of no coverage.
I assume that we would not provide coverage soon
Is it possible to disable the coverage check so it passes?

**Comment by stepansergeevitch** [comment]
Also, it seems like FOSSA fails with an exception

File: .github/workflows/security-scan.yml (hunk 1 above)
Comment (ptiurin) [reply .github/workflows/security-scan.yml:20]: Yeah, once we have the org-wide secrets they'll be upper-case and I will remove this one.

**Comment by ptiurin** [comment]
> Is it possible to disable the coverage check so it passes?

Unfortunately no, sonarqube/cloud does not let us disable the check. It's annoying, but we might need to ignore this check ourselves or re-visit a way to provide coverage information.
We've seen it before being wonky, failing or not failing for no particular reason.

**Comment by ptiurin** [comment]
> Also, it seems like FOSSA fails with an exception

Followed up on IM: Fossa decoration is not working for some reason. I've added the next recommended thing - fossa-test that fails the PR if issues are found. This is not the build itself, it's FOSSA finding problems (which we'll address shortly).

**Comment by ptiurin** [comment]
In fact, both issues Fossa has found are false-positives. I didn't resolve them in order to provide an indication what will happen if such issues are found.

File: .github/workflows/security-scan.yml (hunk 1 above)
Comment (ptiurin) [reply .github/workflows/security-scan.yml]: Fixed so there's no need to change this later.

**Pull request merged**